tinygrad/ops.py

```diff
@@ -32,10 +32,7 @@ def __init__(self, op: Op, src: Tuple[Union[LazyOp, LazyBuffer], ...], arg: Any
     except AttributeError: pass
 
   def __repr__(self): return f"LazyOp(op={self.op}, src={self.src}, arg={self.arg})"
-  def __eq__(self, __value: object) -> bool:
-    if __value.__class__ is not LazyOp: return False
```

Comment:
Is the `cast` here is unnecessary due to the `if` statement previously that check its class type? There's a few more `cast(LazyOp, )` as well and they fall under different condition?

Reply:
The cast here was necessary because `.__class__` was used instead of `isinstance`. Mypy only understands `isinstance`. Without either the cast or `isinstance`, mypy will complain about the attribute access of `__value`.

There are some other places in the code where `isinstance` was replaced by `.__class__ is ...` and then a `cast`, which together is both slower and more complex than just using `isinstance`. I'm slowly undoing those.

Reply:
E.g.

```
 def exec_ast(self, ast:LazyOp, output=None, context=None, **kwargs):
    if TernaryOps.MULACC in self.fxn_for_op and ast.op == ReduceOps.SUM and ast.src[0].__class__ is LazyOp and ast.src[0].op == BinaryOps.MUL:
      ast = LazyOp(TernaryOps.MULACC, cast(LazyOp, ast.src[0]).src, ast.arg)
```

Same thing.

Reply:
I can revert the others, separate MR though.

Reply:
See https://github.com/tinygrad/tinygrad/pull/1694

Reply:
Thanks for the explanation